manifests/daemon_reload.pp

```diff
@@ -0,0 +1,42 @@
+# @summary Run systemctl daemon-reload
+#
+# @api public
+#
+# @param name
+#   A globally unique name for the resource
+#
+# @param enable
+#   Enable the reload exec
+#
+#   * Added in case users want to disable the reload globally using a resource collector
+#
+# @param lazy_reload
+#   Enable a global lazy reload.
+#
+#   * This is meant for cleaning up systems that may have gotten out of sync so no particular
+#     care is taken in trying to actually "fix" things, that would require a custom type that
+#     manipulates the running catalog tree.
+#
+define systemd::daemon_reload (
+  Boolean $enable      = true,
+  Boolean $lazy_reload = false,
+) {
+  if $enable {
+    if $lazy_reload {
```

Comment:
This looks odd in a define. Why would you need multiple instances of this? It feels to me this belongs in a class instead.

Reply:
I'm guessing because you might have to use it more than once for 2 separate services where one of those services also depends on the other??

Reply:
@ekohl You usually don't but, if for some reason, you want to hit *all* declarations of the defined type without having to include an extra class in the defined type (and potentially risk dependency loops), this gives you a way to do it.

Given that the only reason that is there is to try and fix the mess from 2.9.0 to now, it seemed reasonable.

Reply:
Honestly, it would probably work better with a dose of `simplib::dlookup()` but I don't know that the world is ready for that level of fun yet.

Reply:
Still, there is no determination where it shows up. It happens before **all** services, but you have no idea where it shows up. Worst, it also creates a second exec that ends up in the catalog but nothing that refreshes it. It still feels like something I can't wrap my head around. If the `refreshonly` exec is in the `else` part of `if $lazy_reload` it would make more sense to me.

Reply:
@ekohl Did this make sense or do I need to remove the parameter for a sign-off?

Reply:
@ekohl Well, it will show up directly before all of the services :-D.

Honestly, I think I'll just remove it since it's causing issues with getting the fix through.